src/main/java/nl/tudelft/cse1110/andy/execution/step/RunExternalProcessMetaTestsStep.java

```diff
@@ -0,0 +1,82 @@
+package nl.tudelft.cse1110.andy.execution.step;
+
+import nl.tudelft.cse1110.andy.config.DefaultRunConfiguration;
+import nl.tudelft.cse1110.andy.config.MetaTest;
+import nl.tudelft.cse1110.andy.config.RunConfiguration;
+import nl.tudelft.cse1110.andy.execution.Context;
+import nl.tudelft.cse1110.andy.execution.ExecutionStep;
+import nl.tudelft.cse1110.andy.execution.metatest.implementations.ExternalProcessMetaTest;
+import nl.tudelft.cse1110.andy.execution.mode.Action;
+import nl.tudelft.cse1110.andy.grade.GradeCalculator;
+import nl.tudelft.cse1110.andy.result.MetaTestResult;
+import nl.tudelft.cse1110.andy.result.Result;
+import nl.tudelft.cse1110.andy.result.ResultBuilder;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class RunExternalProcessMetaTestsStep implements ExecutionStep {
+
+    @Override
+    public void execute(Context ctx, ResultBuilder result) {
+        RunConfiguration runCfg = ctx.getRunConfiguration();
+
+        int score = 0;
+        int totalWeight = 0;
+
+        try {
+            List<MetaTest> metaTests = runCfg.metaTests();
+            List<MetaTestResult> metaTestResults = new ArrayList<>();
+
+            for (MetaTest abstractMetaTest : metaTests) {
+                if (!(abstractMetaTest instanceof ExternalProcessMetaTest)) {
```

Comment:
I was indeed thinking that the meta tests we have and this one are different things... and the need for an `instanceof` shows that. If you gotta ask the class about which concrete implementation it actually is, then, this means this is not a good abstraction anymore.

Reply:
The main reason I introduced this abstraction is that I wanted to make defining external process meta tests in the configuration just as simple as defining other meta tests:
```java
    @Override
    public List<MetaTest> metaTests() {
        return List.of(
                MetaTest.withExternalProcess(2, "home page does not load",
                        new CommandExternalProcess(
                                "java -jar petclinic-metatest1.jar",
                                "initialised")
                )
        );
    }
```

My initial idea was to have two separate strategies for evaliating meta tests - one which replaces the library and one which replaces the external process. This would eliminate the need for two separate steps. However, this is not possible without running the step twice - before and after terminating the original external process,  since library replacements have to be done before the original external process is killed. (Technically, Selenium assignments don't need to have a library, so it wouldn't make a difference even if it doesn't work for them, but maybe there could be Selenium assignments with library code in the future.)

I'll think about this in the next few days and see if I can figure out another way of executing meta tests with strategies.

Reply:
I think these two things will never be mixed. If it's selenium, then, it's a black box test and there won't ever be library code. Or vice-versa. Can you see one example of both existing together? And anyways, if this forces us to use `instanceof`, I'd say it's not worth merging them. `instanceof` is one of the biggest red flags we can have in code IMHO.

Reply:
I can't think of any use case where both will be used simultaneously.

I have an idea how we can remove `instanceof`. We could use the strategy pattern (in order to remove the need for two separate meta test executors) together with an `executionType` which will be different per meta test type - that execution type can be either `BEFORE_KILLING_EXTERNAL_PROCESS` or `AFTER_KILLING_EXTERNAL_PROCESS`. We can then call the meta test step twice in the pipeline - once before killing the external process, which would execute meta tests of the first type, and once after that, which would run the other meta tests. This would remove the need for `instanceof` and would instead move this logic to the meta test class itself.

Reply:
The problem is that the "before killing the process" type makes no sense at all for the LibraryMetaTests. :/ The external process thingy is leaking to a place it shouldn't

Reply:
Should I then simply run all meta tests after the process is killed (which would work for external meta tests and wouldn't make any difference for library meta tests)? That way we don't need any such complicated logic, and the `instanceof` can easily be replaced by separating the execution strategies.

Reply:
Not sure what you mean (running after the process is killed), but sure, give it a try! If it removes the need for instanceof I'm happier :)

Reply:
I fixed it by moving the execution logic to the meta test children themselves.